fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryView.kt

```diff
@@ -119,6 +120,14 @@ class HistoryView(
         }
     }
 
+    /**
+     * Updates the View with the latest changes to [AppState].
+     */
+    fun update(state: AppState) {
+        historyAdapter.updatePendingDeletionItems(state.pendingDeletionHistoryItems)
+        historyAdapter.notifyDataSetChanged()
+    }
```

Comment:
This wasn't necessarily written with the intent that later additions would use other `AppState` properties. I'm open to change it such that the parameter is `pendingDeletionHistoryItems`, but thought this more closely mirrored the other `update` function which accepts a `HistoryFragmentState` and would help differentiate them and add context of where the update was coming from.

Reply:
ah, gotcha gotcha. I'd normally be a proponent of keeping the parameter dependencies as clear as possible, but since it kinda goes back to `HistoryFragment`, in general, needing a refactor, I'm fine keeping this as-is.